### ISSUE TITLE
application processes consume ~200% CPU combined

### ISSUE THREAD

**Issue by pshomov** [Microsoft/vscode#41557]
```
Version:          Code 1.19.2 (490ef761b76b3f3b3832eff7a588aac891e5fe80, 2018-01-10T15:49:04.682Z)
OS Version:       Darwin x64 17.3.0)
CPUs:             Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)
Memory (System):  16.00GB (0.58GB free)
Load (avg):       2, 2, 2
VM:               0%
Screen Reader:    no

CPU %	Mem MB	Process
   79	    98	code main
    0	    66	   gpu-process
   78	   213	   window (setup-folders.js — ops)
    0	    49	     watcherService
    0	    98	     extensionHost
    0	    98	       electron_node electronForkStart.js tsserver.js
    0	    49	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    82	   shared-process
    0	   197	   window (main.yml — ansible2)
    0	    49	     watcherService
    0	   131	     extensionHost
    0	    33	       electron_node server.js

Workspace Stats:
|  Window (setup-folders.js — ops)
|    Folder (ops): 10 files
|      File types: js(8) json(2)
|      Conf files: package.json(1)
|  Window (main.yml — ansible2)
|    Folder (ansible2): 8307 files
|      File types: py(3407) pyc(3292) json(443) so(127) yml(119) h(97) txt(97)
|                  ps1(57) pem(45) rst(43)
|      Conf files: makefile(1) package.json(1)
```
**Does this issue occur when all extensions are disabled?**: Yes

**Try to reproduce the issue in the VS Code Insider version.**
Attempted this in the Insider edition too with nothing extra installed. Here are my settings 
```
{
    "editor.fontSize": 15
}
```
Same issue:
```
~ » code-insiders --status                             petar@Petars-MacBook-Pro

Version:          Code - Insiders 1.20.0-insider (b36655544f413f67aec02d64ce619d165749b957, 2018-01-15T06:08:20.340Z)
OS Version:       Darwin x64 17.3.0
CPUs:             Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)
Memory (System):  16.00GB (0.08GB free)
Load (avg):       2, 2, 2
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron .

CPU %	Mem MB	   PID	Process
   79	    66	  9194	code-insiders main
    0	    33	  9205	   gpu-process
   77	    98	  9206	   window (docker-compose-services.yml — dev)
    0	    16	  9210	     watcherService
    0	    33	  9211	     extensionHost
    0	    16	  9233	       electron_node jsonServerMain.js 
    0	    33	  9207	   shared-process
    0	   180	 10628	   window (_ascore-backup.yml — ansible2)
    0	    49	 10629	     watcherService
    0	    66	 10630	     extensionHost

Workspace Stats: 
|  Window (docker-compose-services.yml — dev)
|    Folder (dev): 452 files
|      File types: json(274) yml(46) sh(45) json1(30) DS_Store(26) env(17)
|                  js(10) md(2) gitignore(1) log(1)
|      Conf files: package.json(1)
|  Window (_ascore-backup.yml — ansible2)
|    Folder (ansible2): 8309 files
|      File types: py(3407) pyc(3292) json(443) so(127) yml(121) h(97) txt(97)
|                  ps1(57) pem(45) rst(43)
|      Conf files: makefile(1) package.json(1)

------------------------------------------------------------
```

**Opening the VSCode "About" dialog fixes the issue (CPU goes back to pretty much 0) for a while, issue comes back after a while**
I think it is worth mentioning I experience similar issue with Atom too (but not with Slack, so not all Electron-based apps are affected)
This has been going on for a really long time now but your recent improvements in providing details about this issue is making it easier for me to report it.

**Comment by mastrauckas** [comment]
I seem to be having cpu issues with 1.20.0-insider on a macos laptop.  When I bring up code, the fan gets really loud and the laptop gets hot and it drains my battery really fast.  I haven't tested with or without extensions yet.

**Comment by leovarmak** [comment]
Aha. Now I know why my MacBook doesn't last long

**Comment by bpasero** [comment]
@pshomov can you follow https://github.com/Microsoft/vscode/wiki/Performance-Issues

**Comment by mastrauckas** [comment]
My issue seem to be caused from [this](https://github.com/wix/import-cost/issues/47) issue.

**Comment by pshomov** [comment]
@mastrauckas I think it is not related, I have had this issue for much longer. Guessing at least 9 months.

**Comment by pshomov** [comment]
@bpasero I thought I did. I have provided the information from `code --status` and verified I am getting the same issue with extensions disabled.

I should also mention (and I already updated the info in the issue) opening the VSCode "About" dialog fixes the issue (CPU goes back to pretty much 0) for a while, issue comes back after a while

**Comment by bpasero** [comment]
@pshomov there is more to it: https://github.com/Microsoft/vscode/wiki/Performance-Issues#the-rendererwindow-process-consumes-a-lot-of-cpu

**Comment by pshomov** [comment]
@bpasero my bad, will follow up with the rest of the tests

**Comment by shahata** [comment]
Issue is fixed in import-cost version 2.6.2, sorry for the trouble...

**Comment by pshomov** [comment]
@bpasero  @shahata I am not sure how import-cost is relevant to this issue. I have not installed this extension. Can we please put this issue back to open. I am still trying to gather as much evidence as possible to get this bug.

**Comment by bpasero** [comment]
> @bpasero my bad, will follow up with the rest of the tests

So?

**Comment by anklaa** [comment]
I have the same issue. VScode drains my battery.
`
Version:          Code 1.19.3 (7c4205b5c6e52a53b81c69d2b2dc8a627abaa0ba, 2018-01-25T10:25:01.248Z)
OS Version:       Darwin x64 17.3.0)
CPUs:             Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)
Memory (System):  16.00GB (1.46GB free)
Load (avg):       2, 3, 3
VM:               0%
Screen Reader:    no
[main 10:43:14] TypeError: Cannot read property 'mem' of undefined
    at p (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:183:177)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:181:730
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-main/main.js:181:754
    at <anonymous>`

**Comment by pshomov** [comment]
@bpasero Have been running with no custom settings - same issue. It seems opening and selecting anything in the menu fixes the issue (previously though it was only the "About" box but just dropping down the menu seems to help). I will keep the developer tools window open and wait for the issue to happen again. Hope to have something tomorrow

**Comment by pshomov** [comment]
@bpasero Have been launching every VSCode window/folder like this `code --verbose /some/path` in the last few days. The problematic behaviour continues and there is nothing of interest (in my opinion) in the logs. For example in the last 30 mins or so - 
```
[26088:0207/154739.056124:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154739.109712:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.316980:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.368165:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.822297:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[ADDED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.876604:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.921410:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.921796:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[DELETED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.921843:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.972919:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.972972:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154746.973004:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.857894:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.858280:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.858558:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.858959:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.908798:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.908850:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.908882:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154802.908911:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154807.833043:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154807.833381:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c[CHANGED]", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154807.887577:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
[26088:0207/154807.888332:INFO:CONSOLE(249)] "%c[IPC Library: Watcher] %c >> normalized", source: file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js (249)
```

@anklaa Please try to open any of the menu items in VSCode (I go for the "About" box) the next time this happens and see if it fixes it for a while?

**Comment by pshomov** [comment]
Here is also a screenshot of my Activity Monitor when the issue is happening 
<img width="801" alt="screen shot 2018-02-07 at 15 56 43" src="https://user-images.githubusercontent.com/123692/35926304-bc36550c-0c1f-11e8-8523-5e6fd92307b3.png">

**Comment by pshomov** [comment]
@bpasero Here are a set of cpuprofiles which I recorded while this is happening. I had three windows opened so was recording for each one. Hope this is somewhat helpful.


[CPU-20180207T180325.cpuprofile.zip](https://github.com/Microsoft/vscode/files/1704216/CPU-20180207T180325.cpuprofile.zip)
[CPU-20180207T180148.cpuprofile.zip](https://github.com/Microsoft/vscode/files/1704226/CPU-20180207T180148.cpuprofile.zip)
[CPU-20180207T180305.cpuprofile.zip](https://github.com/Microsoft/vscode/files/1704227/CPU-20180207T180305.cpuprofile.zip)

**Comment by pshomov** [comment]
@bpasero Here are the two processes from VSCode that are causing the trouble

```
~/src/as/infra/docker-images(master) » ps auxww | grep 44765
petar            44765  73.9  0.8  5455780 133896   ??  Ss    1:07PM  10:39.60 /Applications/Visual Studio Code.app/Contents/MacOS/Electron ..

~/src/as/infra/docker-images(master) » ps auxww | grep 44777
petar            44777  56.1  1.7  6150028 277136   ??  S     1:07PM   7:08.55 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --type=renderer --js-flags=--nolazy --no-sandbox --primordial-pipe-token=7980170C907202C4F3487530A9541A5E --lang=en-GB --app-path=/Applications/Visual Studio Code.app/Contents/Resources/app --node-integration=true --webview-tag=true --no-sandbox --background-color=#171717 --disable-blink-features=Auxclick --enable-pinch --num-raster-threads=4 --enable-zero-copy --enable-gpu-memory-buffer-compositor-resources --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,34037;0,11,34037;0,12,34037;0,13,3553;0,14,3553;0,15,3553;1,0,3553;1,1,3553;1,2,3553;1,3,3553;1,4,3553;1,5,3553;1,6,3553;1,7,3553;1,8,3553;1,9,3553;1,10,34037;1,11,34037;1,12,34037;1,13,3553;1,14,3553;1,15,3553;2,0,3553;2,1,3553;2,2,3553;2,3,3553;2,4,3553;2,5,3553;2,6,3553;2,7,3553;2,8,3553;2,9,3553;2,10,34037;2,11,34037;2,12,34037;2,13,3553;2,14,3553;2,15,3553;3,0,3553;3,1,3553;3,2,3553;3,3,3553;3,4,3553;3,5,34037;3,6,3553;3,7,3553;3,8,3553;3,9,3553;3,10,3553;3,11,3553;3,12,34037;3,13,3553;3,14,34037;3,15,34037;4,0,3553;4,1,3553;4,2,3553;4,3,3553;4,4,3553;4,5,34037;4,6,3553;4,7,3553;4,8,3553;4,9,3553;4,10,3553;4,11,3553;4,12,34037;4,13,3553;4,14,34037;4,15,34037 --service-request-channel-token=7980170C907202C4F3487530A9541A5E --renderer-client-id=4

```

**Comment by bpasero** [comment]
@pshomov does it reproduce in any folder you open or even when not opening a folder?

**Comment by pshomov** [comment]
@bpasero I am almost always working with folders. It seems it is not connected to any particular folder but just opening a few of them. Here is the latest where the CPU usage is on a folder with 4 files(sorry already on 1.20 but I this issue has been for a really long time so I do not think it matter that much that I got upgraded to the latest)
```
Version:          Code 1.20.0 (c63189deaa8e620f650cc28792b8f5f3363f2c5b, 2018-02-07T17:02:34.244Z)
OS Version:       Darwin x64 17.4.0
CPUs:             Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)
Memory (System):  16.00GB (0.08GB free)
Load (avg):       4, 4, 4
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron ..

CPU %	Mem MB	   PID	Process
  130	   147	 44765	code main
    0	    66	 44776	   gpu-process
    0	   197	 44777	   window (package.json — dev)
    0	    66	 44781	     extensionHost
    0	    33	 44784	       electron_node jsonServerMain.js 
    0	    33	 44782	     watcherService
    0	    33	 44989	     searchService
    0	    82	 44778	   shared-process
    0	   197	 44835	   window (_gocd-agent.yml — ansible2)
    0	    49	 44836	     extensionHost
    0	    33	 44837	     watcherService
  127	   295	 48520	   window (docker-compose-cli.yml — provision-env)
    0	    82	 48521	     extensionHost
    0	   115	 48550	       electron_node electronForkStart.js tsserver.js 
    0	    49	 48552	         electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    33	 48559	       electron_node jsonServerMain.js 
    0	    49	 48522	     watcherService
    0	    49	 48548	     searchService
    0	   197	 49024	   window (Dockerfile (Working Tree) — docker-images)
    0	    66	 49025	     extensionHost
    0	    49	 49026	     watcherService
    0	    49	 49070	     searchService

Workspace Stats: 
|  Window (package.json — dev)
|    Folder (dev): 509 files
|      File types: json(322) yml(48) sh(47) json1(31) DS_Store(27) env(17)
|                  js(11) md(2) ts(2) gitignore(1)
|      Conf files: launch.json(1) package.json(1)
|  Window (_gocd-agent.yml — ansible2)
|    Folder (ansible2): 8316 files
|      File types: py(3407) pyc(3292) json(443) so(127) yml(124) h(97) txt(97)
|                  ps1(57) pem(45) rst(43)
|      Conf files: makefile(1) package.json(1)
|  Window (docker-compose-cli.yml — provision-env)
|    Folder (provision-env): 4 files
|      File types: ts(2) yml(1) js(1)
|      Conf files:
|  Window (Dockerfile (Working Tree) — docker-images)
|    Folder (docker-images): 52 files
|      File types: sh(14) json(12) DS_Store(8) pem(3) md(1) gitignore(1)
|                  dockerignore(1) cast5(1) properties(1) jar(1)
|      Conf files: package.json(1)
```

**Comment by pshomov** [comment]
@bpasero Not sure if it matters I always launch VSCode from the terminal using its cli `code`. I sometime open individual files (again using the cli) but they seem to open in the whatever the foremost VSCode window is.

**Comment by bpasero** [comment]
@pshomov are any other applications running when Code CPU spins? Maybe try to reboot and only run Code to see if it changes anything.

**Comment by pshomov** [comment]
@bpasero I run very consistent set of apps - Safari, Slack, Terminal, VSCode. I have clipboard manager  and 1password running on the side. Will try to turn everything off and just troubleshoot this issue

**Comment by pshomov** [comment]
@bpasero @anklaa  I have great news (at least great for me): I started launching VSCode from the finder and just dropping the folders on the app and I have not seen this problem in a few days. Before the issue would show up in some cases many times per hour.
I can only attribute this problem to a difference in the environment variables. @bpasero my understanding is that VSCode is pretty standalone with its own version of node bundled. Isn't that right? Any ideas what should I troubleshoot?

**Comment by bpasero** [comment]
@pshomov so before you would start from the command line, is that what you are saying?

**Comment by pshomov** [comment]
@bpasero yes, as mentioned previously.

I spoke too soon :( . Maybe I had too many meetings those days but I am back to the same problem after using VSCode for a bit longer. Will try to investigate over the weekend this launching from the Finder vs launching from the cli difference.

**Comment by pshomov** [comment]
@bpasero Any ideas what I should try here more? Here is the latest instance of the issue:

```
Version:          Code 1.20.1 (f88bbf9137d24d36d968ea6b2911786bfe103002, 2018-02-13T15:23:28.607Z)
OS Version:       Darwin x64 17.4.0
CPUs:             Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2700)
Memory (System):  16.00GB (1.74GB free)
Load (avg):       2, 2, 2
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron

CPU %	Mem MB	   PID	Process
   94	    98	 11833	code main
    0	    49	 11835	   gpu-process
    0	   180	 11837	   shared-process
    0	   229	 12092	   window (run.ts — dev)
    0	    98	 12093	     extensionHost
    0	    49	 12168	       electron_node tslintServer.js 
    0	   115	 12169	       electron_node electronForkStart.js tsserver.js 
    0	     0	 12173	         electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    16	 12618	       electron_node jsonServerMain.js 
    0	    16	 12094	     watcherService
    0	   197	 18821	   window (README.md — kafka-setup)
    0	    82	 18823	     extensionHost
    0	    49	 18824	     watcherService
   74	   229	 20806	   window (key.pem.cast5 — library)
    0	    82	 20807	     extensionHost
    0	    49	 20808	     watcherService
    0	   213	 21243	   window (cert.pem — ansible2)
    0	    82	 21244	     extensionHost
    0	    49	 21245	     watcherService

Workspace Stats: 
|  Window (run.ts — dev)
|    Folder (dev): 517 files
|      File types: json(325) yml(48) sh(47) json1(31) DS_Store(28) env(18)
|                  js(9) ts(5) md(3) gitignore(1)
|      Conf files: launch.json(1) package.json(1) tsconfig.json(1)
|                  tslint.json(1)
|  Window (README.md — kafka-setup)
|    Folder (kafka-setup): 14 files
|      File types: sh(5) properties(2) yml(2) md(1)
|      Conf files:
|  Window (key.pem.cast5 — library)
|    Folder (library): 54 files
|      File types: sh(15) json(12) DS_Store(8) pem(6) cast5(2) properties(1)
|                  jar(1)
|      Conf files: package.json(1)
|  Window (cert.pem — ansible2)
|    Folder (ansible2): 8323 files
|      File types: py(3407) pyc(3292) json(443) yml(128) so(127) txt(97) h(97)
|                  ps1(57) pem(45) rst(43)
|      Conf files: makefile(1) package.json(1)
```

**Comment by pshomov** [comment]
@bpasero could this be some hardware issue with the GPU?

**Comment by bpasero** [comment]
@pshomov does it reproduce if you run "`code --disable-gpu`"?

**Comment by pshomov** [comment]
@bpasero Baam! Found the issue - it has nothing to do with VSCode and everything to do with an app I have been using for years and years and really forgotten it is a third-party app - https://manytricks.com/witch/. It is window switcher for macOs. It just occurred to me that it could be an issue after I noticed the problem occurs after I switch between windows in VSCode.

Sorry for wasting your time @bpasero, hope this helps someone else so it was not just all for me 😞
